tests/unit/orcid/test_orcid_builder.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from inspire_utils.date import PartialDate
+from inspirehep.modules.orcid import OrcidBuilder
+from lxml import etree
+
+
+def xml_parse(xml_string):
+    """Parse an ``xml_string`` into XML."""
+    parser = etree.XMLParser(remove_blank_text=True)
+    return etree.fromstring(xml_string, parser)
+
+
+def xml_compare(result, expected):
+    """Assert two XML nodes equal."""
+    assert etree.tostring(result, pretty_print=True) == etree.tostring(expected, pretty_print=True)
```

Comment:
Mh, comparing XMLs by comparing their string representations is trickier than it looks because of things than attribute ordering, entity dereferencing and namespace representation (more painful details: https://www.w3.org/2001/tag/doc/xmlChunkEquality.html). I'm not sure if the `pretty_print=True` argument takes care of all these...